src/dynstore.rs

```diff
@@ -1,32 +1,43 @@
-use super::{dynamodbstore, memorystore, readcache, redisstore, Arg, AtomicWriteOperation, BatchOperation, Result, Value};
+use super::{memorystore, readcache, Arg, AtomicWriteOperation, BatchOperation, Result, Value};
 use std::{collections::HashMap, ops::Bound};
 
 #[derive(Clone)]
+#[non_exhaustive]
 pub enum Backend {
     Memory(memorystore::Backend),
-    Redis(redisstore::Backend),
-    DynamoDB(dynamodbstore::Backend),
+    #[cfg(feature = "redis")]
+    Redis(crate::redisstore::Backend),
+    #[cfg(feature = "rusoto")]
+    Rusoto(crate::rusotostore::Backend),
     ReadCache(Box<readcache::Backend<Backend>>),
 }
 
+/// Delegates to each supported backend via identical expressions.
+///
+/// Rust doesn't allow macros to reference variables not passed into them or exprs passed into
+/// macros to reference variables defined in the macro. Thus, `self` and `backend` must be supplied
+/// as parameters.
+macro_rules! dispatch {
+    ($self:ident, $backend:ident, { $expansion:expr }) => {
+        match $self {
+            Self::Memory($backend) => $expansion,
+            #[cfg(feature = "redis")]
+            Self::Redis($backend) => $expansion,
+            #[cfg(feature = "rusoto")]
+            Self::Rusoto($backend) => $expansion,
+            Self::ReadCache($backend) => $expansion,
+        }.await
+    }
+}
+
 #[async_trait]
 impl super::Backend for Backend {
     async fn get<'a, K: Into<Arg<'a>> + Send>(&self, key: K) -> Result<Option<Value>> {
-        match self {
-            Self::Memory(backend) => backend.get(key).await,
-            Self::Redis(backend) => backend.get(key).await,
-            Self::DynamoDB(backend) => backend.get(key).await,
-            Self::ReadCache(backend) => backend.get(key).await,
-        }
+        dispatch!(self, backend, { backend.get(key) })
```

Comment:
Much nicer! :D

Reply:
> My only hesitation is I think it'd be a little clearer to keep "DynamoDB" in the variant / module names (e.g. "RusotoDynamoDB"), but I can go either way.

I debated this a bit myself and first chose brevity, but you're right: this doesn't get spelled that often, and it's better to be clear. Now says `rusoto_dynamodbstore` / `RusotoDynamoDB`. Coming soon: `aws_sdk_dynamodbstore` / `AwsSdkDynamoDB`. PTAL?